### DISCUSSION TITLE
Bouncy vehicle

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#96] [General]
Hmm, I have tried to replicate the wheeled vehicle example. In my case, however, the vehicle bounces off the wall aggressively when coming at it at a sharp angle. Even velocity is very low. I tried different wheel and suspension sizes/params. Making the params more tighter, towards go-kart style, the bounce becomes smaller, but still there. If the params more lose, like in the example or even higher, towards an offroad style suspension, then the bounce is even more dramatic. Toggling the suspension force point doesn't change much. The friction and restitution for the environment and vehicle are at zero. Any ideas what may cause it?

https://github.com/jrouwe/JoltPhysics.js/assets/5677782/febc518a-d2f9-4914-800a-9ac820d468c9

**Comment by jrouwe** [comment]
You're not drawing your wheels, but from the looks of it they are touching the wall first. Typically if a wheel comes in contact with a vertical wall, the suspension shoots up (remember the wheel is just a collision check and not a physical thing so if it starts penetrating it will immediately jump up) and causes a jolt. You can try moving your wheels inwards so that the wall touches the vehicle body first, or you can add some protective collision around the side of the car.

**Comment by LeXXik** [comment]
You were right! The wheels were the culprit. Adding some temp renders for them showed they are sticking out of the body collision shape and react to wall, pushing the vehicle:

https://github.com/jrouwe/JoltPhysics.js/assets/5677782/5cce607d-58a3-4121-85ff-dc3781317f62

I'm also curious if there is a way to debug draw the wheels/suspensions. I mean similarly to how we debug draw the shapes. Probably no easy way to do that, otherwise the same could have been used for drawing constraints that connect bodies.

**Reply by jrouwe** [reply]
With https://github.com/jrouwe/JoltPhysics/pull/889 the wheels can be drawn, but unfortunately in JoltPhysics.js the DebugRenderer is disabled since Jolt is compiled in Distribution mode. Technically it would be possible to create a version of the library that enables debug drawing but that means that the number of artifacts double and it also adds complications to the idl file since the DebugRenderer would have to be conditionally defined (and I don't think idl supports this). So for now the best way is to draw the wheels yourself.
